test/torch/tensors/test_fv.py

```diff
@@ -691,8 +691,15 @@ def test_fv_relin(val):
     decryptor = Decryptor(ctx, keys[0])  # keys[0] = secret_key
     evaluator = Evaluator(ctx)
 
-    op = encryptor.encrypt(encoder.encode(val))
-    temp_prod = evaluator.mul(op, op)
+    op1 = encryptor.encrypt(encoder.encode(val1))
+    op2 = encryptor.encrypt(encoder.encode(val2))
+    temp_prod = evaluator.mul(op1, op2)
     relin_prod = evaluator.relin(temp_prod, relin_key)
     assert len(temp_prod.data) - 1 == len(relin_prod.data)
-    assert val * val == encoder.decode(decryptor.decrypt(relin_prod))
+    assert val1 * val2 == encoder.decode(decryptor.decrypt(relin_prod))
+
+    with pytest.raises(Warning):
+        evaluator.relin(op1, relin_key)  # Ciphertext size 2
+
+    with pytest.raises(Exception):
+        evaluator.relin(evaluator.mul(temp_prod, val1), relin_key)  # Ciphertext size 4
```

Comment:
We doesn't need to run this on every values, I think you can separate this into a different test with a single value